### DISCUSSION TITLE
Is there one default offset of one sub addrmap?

### DISCUSSION THREAD

**Discussion by peidongbin** [SystemRDL/systemrdl-compiler#175] [Q&A] [answered]
```
addrmap sub_map0{
    default regwidth = 32;
    <% for($i = 0; $i < 76; $i = $i +1 ){ %>
        reg  {
            field {} a[5];
        } reg_<%=$i%>;
    <% }%>
};
addrmap sub_map1{
    default regwidth = 32;
    <% for($i = 0; $i < 76; $i = $i +1 ){ %>
        reg  {
            field {} a[5];
        } reg_<%=$i%>;
    <% }%>
};
addrmap top{
 sub_map0 sub0;
 sub_map1 sub1;
};
```
Here is one example , i found the addr of sub1 not start with the end of sub0.  is there one default of sub map ?

**Comment by amykyta3** [comment, marked as answer]
The default address packing behavior is to align objects to offsets based on their size, rounded up to the next power-of-two. Therefore the `sub1` instance is allocated to an address of 0x200.
If you want tight packing, you can use the `addressing = compact;` property.
Also, in your example there is no need to use Perl preprocessing. Instead you can use an array of registers:

```
addrmap sub_map0{
    default regwidth = 32;
    reg {
        field {} a[5];
    } myreg[76];
};
addrmap sub_map1{
    default regwidth = 32;
    reg {
        field {} a[5];
    } myreg[76];
};
addrmap top{
    addressing = compact;
    sub_map0 sub0;
    sub_map1 sub1;
};
```

Alternatively, you can assign the addresses explicitly:
```
addrmap top{
    sub_map0 sub0 @ 0x000;
    sub_map1 sub1 @ 0x130;
};
```
I prefer explicit address allocation since it is more obvious how addresses will be allocated.

**Reply by peidongbin** [reply]
thank ！
